### ISSUE TITLE
Can't schedule a background task to run with a delay, or period, of 50 days or more.

### ISSUE THREAD

**Issue by JasonElkin** [umbraco/Umbraco-CMS#16936]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

13.4.1

### Bug summary

`IRecurringBackgroundJob` doesn't support TimeSpans of 4294967295 milliseconds or greater (about 49.7 days).

### Specifics

I understand that this is a limitation of the underlying implementation of System.Threading.Timer, but I don't think that limitation should apply to Umbraco's scheduled tasks.

### Steps to reproduce

1. Create a new IRecurringBackgroundJob as per the docs: https://docs.umbraco.com/umbraco-cms/reference/scheduling#minimal-example
2. Set the Period as follows:
```csharp
 public TimeSpan Period { get => TimeSpan.FromDays(50); }
 ```

### Expected result / actual result

Umbraco Schedules a task for 50 days in the future.

**Comment by JasonElkin** [comment]
I think that the Timer API is too low level to properly serve the kind of scheduled tasks that a web application might need to handle. 

We should really have an extra abstraction layer responsible for scheduling tasks independently of their invocation, rather than doing both with the System.Threading.Timer.

An abstraction for scheduling would also make it possible to support scheduling tasks in "human" time, using DateTime or cron expressions - which currently can't be supported.

**Comment by enkelmedia** [comment]
@JasonElkin I tend to use Hangfire for situations like this, and this great dashboard https://marketplace.umbraco.com/package/cultiv.hangfire from @nul800sebastiaan 

Is that an option?

**Comment by JasonElkin** [comment]
@enkelmedia yeah, perhaps this is in part a docs/naming issue and Umbraco doesn't actually have task "scheduling" built in.

In fairness, the way things are named in code doesn't really point to "scheduling". My frustration is that the implementation is so close to being able to run tasks on a schedule that it's odd that we don't have that functionality built in.
